features/steps/steps.py

```diff
@@ -35,6 +28,19 @@ def when_i_run_command(context, command, user_spec):
     )
     context.process = process
 
+@when('I run `{command}` "{token}" {user_spec}')
+def when_i_run_attach_token(context, command, user_spec, token=None):
```

Comment:
The error you see when trying to run tox -e behave tells you that this when clause is not unique when compared to 
@when("I run `{command}` {user_spec}")

```



What this means is your text format of of the @when string needs to be more unique in format, 
that's why I introduced a more unique decorator format @when("I attach with token `{token_name}` {user_spec}")
This pattern will not possibly match the @when function above

Full diff here with a few points:  https://paste.ubuntu.com/p/NfRhmKWVdr/

Reply:
Ok I've changed it, but I kept the `token` info from  `UAClientBehaveConfig` class. Please, See if you think it is ok or not.

Reply:
@patriciadomin It's hard for me to know exactly how you are going to use the context.token until you push of suggest the rest of the context around why context.token is needed instead of context.config.token 

Generally, as you see @OddBloke added the UAClientBehaveConfig class to features/environment.py. That class intent is to encapsulate and simplify any reference to external environment configuration in a single place/instance and surface those config options as instance attributes to simplify tests throughout the rest of the behave tests. this way behave steps can use context.config.reuse_image or context.config.image_name etc after the initial UAClientBehaveConfig was set up.  So I think we still need some breadcrumbs of contract config information inside UAClientBehaveConfig so that at least .from_environ() method can print out the config options provided to the behave test run to ease in our debugging failed test runs.



I think it could be okay if you are keeping around context.token for ease of reference.  but I'll have to see the rest of the context around your approach once you git push your next changeset to this branch. 

Another alternative could be that you extend UAClientBehaveConfig to surface a userdata property which would act just like behave's default context.config.userdata, then any steps could reference context.config.userdata as they see fit, just like stock behave tests.

```@property
def userdata(cls):
    return cls.userdata
```